File: project-04-dynamic-search/code-java/src/main/java/com/example/demo/service/ProductSearchService.java
package com.example.demo.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.example.demo.dto.ProductDto;
import com.example.demo.dto.ProductSearchRequestDto;
import com.example.demo.repository.ProductSearchRepository;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class ProductSearchService {

    private final ProductSearchRepository productSearchRepository;

    public List<ProductDto> searchProducts(ProductSearchRequestDto request) {
        return productSearchRepository.searchProducts(request);
    }
}

File: project-04-dynamic-search/code-kotlin/src/main/kotlin/com/example/demo/service/ProductSearchService.kt
package com.example.demo.service

import com.example.demo.dto.ProductDto
import com.example.demo.dto.ProductSearchRequestDto
import com.example.demo.repository.ProductSearchRepository
import org.springframework.stereotype.Service

@Service
class ProductSearchService(
    private val productSearchRepository: ProductSearchRepository
) {

    fun searchProducts(request: ProductSearchRequestDto): List<ProductDto> {
        return productSearchRepository.searchProducts(request)
    }
}
